### PR TITLE
[NO QA][CardSettings] replace corporateCardList with cardList

### PR DESCRIPTION
**Pull request by grgia**
### Details

`cardList` was split into fundList and `corporateCardList` [[PR](https://github.com/Expensify/App/issues/23560)] in the initial design of Expensify Card Settings, but we decided to keep the name `cardList`. 

`fundList` stores cards from the `funds` table 
`cardList` stores cards from the `cards` table

### Fixed Issues

$ https://github.com/Expensify/App/issues/25940

### Tests

- [ ] Verify that no errors appear in the JS console
- [ ] NO QA, key not yet used

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] NO QA, key not yet used

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by grgia** [comment]
Removing C+ as there's no QA

**Review by rushatgabhane — APPROVED** [review]

File: src/ONYXKEYS.ts [hunk 1]
```diff
@@ -327,6 +327,7 @@ type OnyxValues = {
     [ONYXKEYS.WALLET_TERMS]: OnyxTypes.WalletTerms;
     [ONYXKEYS.BANK_ACCOUNT_LIST]: Record<string, OnyxTypes.BankAccount>;
     [ONYXKEYS.FUND_LIST]: Record<string, OnyxTypes.Card>;
+    [ONYXKEYS.CARD_LIST]: Record<string, OnyxTypes.Card>;
```
Comment (MonilBhavsar) [inline src/ONYXKEYS.ts:330]: We need different structure for both cards, no?
I mean OnyxTypes.Fund and OnyxTypes.Card

**Comment by MonilBhavsar** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

File: src/ONYXKEYS.ts [hunk 2]
```diff
@@ -139,11 +139,11 @@ const ONYXKEYS = {
     /** The user's bank accounts */
     BANK_ACCOUNT_LIST: 'bankAccountList',
 
-    /** The user's credit cards (renamed from cardList) */
+    /** The user's credit / P2P cards */
```
Comment (flodnv) [inline src/ONYXKEYS.ts:142]:
```suggestion
    /** The user's cash card and imported cards */
```

As discussed, P2P cards should go in fundList

File: src/ONYXKEYS.ts (hunk 1 above)
Comment (flodnv) [reply src/ONYXKEYS.ts:330]: I am pretty sure that's right, yes

Comment (flodnv) [reply src/ONYXKEYS.ts:330]: Also, why did we not have this line for `ONYXKEYS.CORPORATE_CARD_LIST`?

**Review by flodnv — CHANGES_REQUESTED** [review]

File: src/ONYXKEYS.ts (hunk 1 above)
Comment (grgia) [reply src/ONYXKEYS.ts:330]: > Also, why did we not have this line for ONYXKEYS.CORPORATE_CARD_LIST?

I'm not sure, it was added before the TS migration

Comment (grgia) [reply src/ONYXKEYS.ts:330]: Pushed a fix

**Comment by grgia** [comment]
I don't know why there's conflicts I just pulled and merged main

File: src/types/onyx/Fund.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {ValueOf} from 'type-fest';
+import CONST from '../../CONST';
+
+type Fund = {
```
Comment (grgia) [inline src/types/onyx/Fund.ts:4]: Not sure how in depth TS definitions should go, but if this is eventually a list of slightly different payment card types, should I specify these to be "Expensify Card" specific and have Fund be a union of slightly different card object shapes? 


cc @hayata-suenaga on TS question

Comment (grgia) [reply src/types/onyx/Fund.ts:4]: Kind of a pre-optimization though, we could also determine when we get to it

**Comment by grgia** [comment]
@flodnv @MonilBhavsar ready for review!

File: src/types/onyx/index.ts [hunk 4]
```diff
@@ -24,6 +24,7 @@ import WalletAdditionalDetails from './WalletAdditionalDetails';
 import WalletTerms from './WalletTerms';
 import BankAccount from './BankAccount';
 import Card from './Card';
+import Fund from './Fund';
```
Comment (grgia) [inline src/types/onyx/index.ts:27]: oops need to swap these names 😅

File: src/ONYXKEYS.ts [hunk 5]
```diff
@@ -142,11 +142,11 @@ const ONYXKEYS = {
     /** The user's bank accounts */
     BANK_ACCOUNT_LIST: 'bankAccountList',
 
-    /** The user's credit cards (renamed from cardList) */
+    /** The user's cash card and imported cards */
```
Comment (flodnv) [inline src/ONYXKEYS.ts:145]: fundList: The user's payment and P2P cards

cardList: The user's cash card and imported cards (including the Expensify Card)

File: src/types/onyx/Fund.ts (hunk 3 above)
Comment (grgia) [reply src/types/onyx/Fund.ts:23]: This conversation should move under "Card.ts", I swapped the types

File: src/types/onyx/Card.ts [hunk 6]
```diff
@@ -1,32 +1,17 @@
...
-
-    cardYear?: number;
-    created?: string;
-    currency?: string;
-    fundID?: number;
-};
+import {ValueOf} from 'type-fest';
+import CONST from '../../CONST';
 
 type Card = {
```
Comment (flodnv) [inline src/types/onyx/Card.ts:4]: I don't think any of them are (currently) optional?

File: src/types/onyx/Fund.ts (hunk 3 above)
Comment (flodnv) [reply src/types/onyx/Fund.ts:23]: Once again here, these are not Expensify Cards 😄

File: src/types/onyx/Fund.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    isBillingCard?: boolean;
+    isP2PDebitCard?: boolean;
+};
+
+type AccountData = {
+    additionalData?: AdditionalData;
+    addressName?: string;
+    addressState?: string;
+    addressStreet?: string;
+    addressZip?: number;
```
Comment (flodnv) [inline src/types/onyx/Fund.ts:11]: Here too, I feel like most of these aren't optional...

**Review by flodnv — CHANGES_REQUESTED** [review]

File: src/types/onyx/Fund.ts (hunk 7 above)
Comment (grgia) [reply src/types/onyx/Fund.ts:11]: I didn't modify these , just moved what was in the old Card.ts file into Fund.ts

File: src/types/onyx/Fund.ts (hunk 3 above)
Comment (grgia) [reply src/types/onyx/Fund.ts:23]: this thread was started before [this](https://github.com/Expensify/App/pull/25939/commits/a290a6f8ffb974944eefdb3260b84d2720c75202) commit, sorry for the confusion!

File: src/types/onyx/Fund.ts (hunk 7 above)
Comment (flodnv) [reply src/types/onyx/Fund.ts:11]: I get that, but that's not really a good reason to not fix issues with code 😄

**Review by flodnv — APPROVED** [review]
Approving as I'm about to go ooo and this seems fine, but it would be nice to fix my comment about the Fund type

File: src/types/onyx/Card.ts [hunk 8]
```diff
@@ -1,32 +1,17 @@
...
+    cardID: number;
+    state: number;
+    bank: string;
+    availableSpend: number;
+    domainName: string;
+    maskedPan: string;
+    isVirtual: boolean;
+    fraud: ValueOf<typeof CONST.EXPENSIFY_CARD.FRAUD_TYPES>;
+    cardholderFirstName: string;
+    cardholderLastName: string;
```
Comment (MonilBhavsar) [inline src/types/onyx/Card.ts:14]: Would this be optional, or always available?

Comment (grgia) [reply src/types/onyx/Card.ts:14]: right now always available

**Comment by grgia** [comment]
@MonilBhavsar bump on review, I'd like to merge this today if possible

**Review by MonilBhavsar — APPROVED** [review]

**Pull request merged**